llama-index-core/llama_index/core/query_engine/sql_join_query_engine.py

```diff
@@ -282,26 +288,49 @@ def _query_sql_other(self, query_bundle: QueryBundle) -> RESPONSE_TYPE:
             print_text(f"query engine response: {other_response}\n", color="pink")
         logger.info(f"> query engine response: {other_response}")
 
-        response_str = self._llm.predict(
-            self._sql_join_synthesis_prompt,
-            query_str=query_bundle.query_str,
-            sql_query_str=sql_query,
-            sql_response_str=str(sql_response),
-            query_engine_query_str=new_query.query_str,
-            query_engine_response_str=str(other_response),
-        )
-        if self._verbose:
-            print_text(f"Final response: {response_str}\n", color="green")
-        response_metadata = {
-            **(sql_response.metadata or {}),
-            **(other_response.metadata or {}),
-        }
-        source_nodes = other_response.source_nodes
-        return Response(
-            response_str,
-            metadata=response_metadata,
-            source_nodes=source_nodes,
-        )
+        if self._streaming:
+            response_str = self._llm.stream(
+                self._sql_join_synthesis_prompt,
+                query_str=query_bundle.query_str,
+                sql_query_str=sql_query,
+                sql_response_str=str(sql_response),
+                query_engine_query_str=new_query.query_str,
+                query_engine_response_str=str(other_response),
+            )
+            if self._verbose:
+                print_text(f"Final response: {response_str}\n", color="green")
```

Comment:
printing this will consume the stream I guess lol

Reply:
surprisingly this doesn't do it, but if you try to print a `StreamingResponse` object, the stream gets consumed. I have raised a issue for the same at #13893

Reply:
If it's the final response, why do we need to print it? It will be returned to the user

Reply:
so do you want me to remove this?

Reply:
Yea

Reply:
done